### ISSUE TITLE
SnapSheet stuck on last SnapPosition

### ISSUE THREAD

**Issue by Djcharles26** [AdamJonsson/snapping_sheet#10]
The above snapSheet is getting stucked on the last SnapPosition, and doesn't return to any other SnapPosition

**Comment by AdamJonsson** [comment]
Thanks for posting an issue.

Can you send the code of an example where the issue occurs?

**Comment by Djcharles26** [comment]
Yes of course, its here: 
```dart
SnappingSheet(
                          snappingSheetController: _controller,
                          onSnapBegin: (){
                            if(this._controller.currentSnapPosition.positionPixel <= h * 0.12)
                              Provider.of<UI>(context, listen: false).changeSnapPosition(this._controller.currentSnapPosition);
                          },
                          snapPositions: [
                            SnapPosition(
                              positionPixel: 0,
                            ),
                            SnapPosition(
                              positionPixel: h*0.12,
                            ),
                            SnapPosition(
                              positionPixel: h*0.8,
                            ),
                          ] ,
                          initSnapPosition: SnapPosition(
                            positionPixel: 0
                          ) ,
                          lockOverflowDrag: true,
);
```

If I put lockOverflowDrag as true, it get stucks in last but not in the other to snap positions.

**Comment by AdamJonsson** [comment]
Hi again! Seems that I can not replicate the issue that you are having. We can check the following to easier find the problem:
* Is the problem still occurring when the code in the onSnapBegin callback is removed, if not, what does it do?
* What platform are you using, Android or IOS, a real device or an emulator?
* Are you using the latest version of the snapping sheet?
* What widgets are you inserting in the sheetBelow, grabbing, and sheetAbove parameters?

**Comment by densa** [comment]
It happens when use last SnapPosition with positionPixel and lockOverflowDrag - true

**Comment by proformance** [comment]
I can confirm that I'm also experiencing this issue, with lockOverflowDrag = true it only works once if using positionPixel, if I replace positionPixel with positionFactor it works like a charm.

- Is the problem still occurring when the code in the onSnapBegin callback is removed, if not, what does it do?
- *N/A*
- What platform are you using, Android or IOS, a real device or an emulator?
- *iOS, emulator*
- Are you using the latest version of the snapping sheet?
- *Yes, 2.0.2
- What widgets are you inserting in the sheetBelow, grabbing, and sheetAbove parameters?
- *A simple container with height*

**Comment by AdamJonsson** [comment]
This should be fixed in the new 3.0.0 version of the package. Unfortunately, the update is a complete refactoring of the project with many different renamings of objects and parameters. So it is some work involved if one wants to update to the newest version
